hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/BinaryCompatibilityFileGenerator.java

```diff
@@ -449,7 +432,7 @@ public static void main(String[] args) throws IOException {
 
 
 {
-    ClientMessage clientMessage = MapTryPutCodec.encodeRequest(    aString ,    aData ,    aData ,    aLong ,    aLong ,    aLong   );
+    ClientMessage clientMessage = MapTryPutCodec.encodeRequest(    aString ,    aData ,    aData ,    aLong ,    aLong ,    aBoxedLong   );
```

Comment:
I do not see `tryPut` call with ttl or maxIdle in IMap. tryPut only has `timeout`.

Either tryPut in IMap is missing new maxIdle or MapTryPutMessageTask is missing version checks and handling of the new parameter.

Reply:
The `tryPut` uses the timeout as a waiting period for the `try`. Not as `ttl`.

Reply:
Yes, that is correct. What I mean is IMap.tryPut has neither ttl nor maxIdle but we add maxIdle to its codec? I do not see that parameter used anywehere

Reply:
Absolutely right. Good catch, Thanks
Thats a client-protocol fix.

Reply:
client protocol fix affects `tryPutInternal` in here too. `maxIdle` and `maxIdleUnit` on its contract are useless.